tests/Invoke-DbaDiagnosticQuery.Tests.ps1

```diff
@@ -1,16 +1,102 @@
-$CommandName = $MyInvocation.MyCommand.Name.Replace(".Tests.ps1", "")
+# test ouput directory to confirm creation of test files
+$commandname = $MyInvocation.MyCommand.Name.Replace(".Tests.ps1", "")
 Write-Host -Object "Running $PSCommandpath" -ForegroundColor Cyan
-. "$PSScriptRoot\constants.ps1"
+. (Join-path (($PSScriptRoot, '.\' -ne '')[0]) "constants.ps1")  #if running in interactive console, cd to folder and you can run this without error
+write-host "loaded constants"
 
 Describe "$CommandName Integration Tests" -Tags "IntegrationTests" {
-    Context "Verifying output" {
+    BeforeAll {
+		$script:PesterOutputPath = Join-Path ([System.IO.Path]::GetTempPath()) ([System.IO.Path]::GetRandomFileName())
+		[io.directory]::CreateDirectory($script:PesterOutputPath) > $null
+		write-verbose "Temporary output location for test: $script:PesterOutputPath"
+
+		$database = "dbatoolsci_frk_$(Get-Random)"
+		$database2 = "dbatoolsci_frk_$(Get-Random)"
+        $server = Connect-DbaInstance -SqlInstance $script:instance2
+        $server.Query("CREATE DATABASE $database")
+        $server.Query("CREATE DATABASE $database2")
+        write-verbose "Created Test Databases: $database , $database2 on $script:instance2"
```

Comment:
do not clutter pester tests with verbosity

Reply:
you debug when you code, but you don't code to debug. you code to avoid ANY users debugging.
if you watch at ANY other test, no write-verbose is involved. If you need, use write-debug instead but, again, there's no place for any "write" in pester, as it reports already pretty much everything you need already

Reply:
259 test files are not enough ?! ;-) jokes aside, I think the only pointer would be to read, digest and understand https://github.com/sqlcollaborative/dbatools/blob/development/contributing.md#tests . (think "all code, and to be properly ___serious___ test code too, will be run on users' environment"). Debugging just pleases your daemons when you're building the solution, but it will quickly escalate to "nagging" when:
- niphlod reads the appveyor log to see why a build failed
- the early (but __serious__ ) adopter shows its colleagues dbatools works in their environment, too, running all or part of the test suite

Reply:
Ok, trying to learn here, appreciate the help. I get that the other test files are great examples, I meant more on the design philosophy of what you said, "you debug when you code, but you don't code to debug. you code to avoid ANY users debugging.". I figured I'd learn from this first contribution :-) Working on changes now

Reply:
everyone has his/her own method, the helpful bit is to grasp the "agenda"/"goals".